crates/ra_hir/src/ty/infer.rs

```diff
@@ -198,12 +214,52 @@ impl<'a, D: HirDatabase> InferenceContext<'a, D> {
             obligations: Vec::default(),
             return_ty: Ty::Unknown, // set in collect_fn_signature
             trait_env: lower::trait_env(db, &resolver),
+            coerce_unsized_map: Self::init_coerce_unsized_map(db, &resolver),
             db,
             body,
             resolver,
         }
     }
 
+    fn init_coerce_unsized_map(
```

Comment:
to be honest, I'd prefer the hard-coded list of impls for now, iuntil we're able to use Chalk to do this properly.

Reply:
`std` have quite a lot of impls. Besides pointers and references, we still have `std::cell::{Cell, RefCell, UnsafeCell, Ref[Mut]}`, `std::ptr::NonNull`, `std::pin::Pin`, `std::boxed::Box`, `std::rc::{Rc, Weak}`, `std::sync::{Arc, Weak}`.

About 12 impls in total. Should we hard-code all of them here? (And it requires more "known names")